src/Sylius/Bundle/SettingsBundle/Manager/SettingsManager.php

```diff
@@ -101,20 +102,30 @@ public function __construct(
     /**
      * {@inheritdoc}
      */
-    public function loadSettings($namespace, $ignoreUnknown = true)
+    public function load($schemaAlias, $namespace = null, $ignoreUnknown = true)
     {
-        if (isset($this->resolvedSettings[$namespace])) {
-            return $this->resolvedSettings[$namespace];
+        $schema = $this->schemaRegistry->get($schemaAlias);
+
+        $resolver = $this->defaultResolver;
+
+        if ($this->resolverRegistry->has($schemaAlias)) {
+            $resolver = $this->resolverRegistry->get($schemaAlias);
         }
 
-        if ($this->cache->contains($namespace)) {
-            $parameters = $this->cache->fetch($namespace);
-        } else {
-            $parameters = $this->getParameters($namespace);
-            $this->cache->save($namespace, $parameters);
+        // try to resolve settings for schema alias and namespace
+        $settings = $resolver->resolve($schemaAlias, $namespace);
+
+        if (!$settings) {
+            $settings = $this->settingsFactory->createNew();
+            $settings->setSchemaAlias($schemaAlias);
         }
 
-        $schema = $this->schemaRegistry->getSchema($namespace);
+        // map parameters to a plain php array
+        $parameters = $settings->getParameters()
```

Comment:
My preference would go to letting `Settings` extend `ArrayObject` or implementing `ArrayAccess` etc. by itself and simply map `parameters` as an array in the Doctrine mapping. The API to work with settings would then be:

```php
// set parameter
$settings["new_parameter"] = "Hello";

// get plain array, to be used when transforming etc.
$parameters = $settings->getArrayCopy();

// replace the transformed parameters
$settings->exchangeArray($parameters);
```